targets/TARGET_STM/PinNamesTypes.h

```diff
@@ -90,15 +90,30 @@ extern "C" {
 #define STM_PIN_ANALOG_CONTROL(X)   (((X) >> STM_PIN_AN_CTRL_SHIFT) & STM_PIN_AN_CTRL_MASK)
 
 #define STM_PIN_DEFINE(FUNC_OD, PUPD, AFNUM)  ((int)(FUNC_OD) |\
-                          ((PUPD  & STM_PIN_PUPD_MASK) << STM_PIN_PUPD_SHIFT) |\
-                          ((AFNUM & STM_PIN_AFNUM_MASK) << STM_PIN_AFNUM_SHIFT))
+						  ((STM_PIN_SPEED_MASK   & STM_PIN_SPEED_MASK) << STM_PIN_SPEED_SHIFT) |\
+                          (((PUPD)  & STM_PIN_PUPD_MASK) << STM_PIN_PUPD_SHIFT) |\
+                          (((AFNUM) & STM_PIN_AFNUM_MASK) << STM_PIN_AFNUM_SHIFT))
+
+#define STM_PIN_DEFINE_SPEED(FUNC_OD, PUPD, AFNUM, SPEEDV)  ((int)(FUNC_OD) |\
+						  (((SPEEDV)   & STM_PIN_SPEED_MASK) << STM_PIN_SPEED_SHIFT) |\
+                          (((PUPD)  & STM_PIN_PUPD_MASK) << STM_PIN_PUPD_SHIFT) |\
+                          (((AFNUM) & STM_PIN_AFNUM_MASK) << STM_PIN_AFNUM_SHIFT))
 
 #define STM_PIN_DEFINE_EXT(FUNC_OD, PUPD, AFNUM, CHAN, INV) \
                                             ((int)(FUNC_OD) |\
-                       ((PUPD   & STM_PIN_PUPD_MASK) << STM_PIN_PUPD_SHIFT) |\
-                       ((AFNUM  & STM_PIN_AFNUM_MASK) << STM_PIN_AFNUM_SHIFT) |\
-                       ((CHAN   & STM_PIN_CHAN_MASK) << STM_PIN_CHAN_SHIFT) |\
-                       ((INV    & STM_PIN_INV_MASK) << STM_PIN_INV_SHIFT))
+                       ((STM_PIN_SPEED_MASK   & STM_PIN_SPEED_MASK) << STM_PIN_SPEED_SHIFT) |\
+                       (((PUPD)   & STM_PIN_PUPD_MASK) << STM_PIN_PUPD_SHIFT) |\
+                       (((AFNUM)  & STM_PIN_AFNUM_MASK) << STM_PIN_AFNUM_SHIFT) |\
+                       (((CHAN)   & STM_PIN_CHAN_MASK) << STM_PIN_CHAN_SHIFT) |\
+                       (((INV)    & STM_PIN_INV_MASK) << STM_PIN_INV_SHIFT))
+
+#define STM_PIN_DEFINE_EXT2(FUNC_OD, PUPD, AFNUM, CHAN, INV, SPEEDV) \
```

Comment:
Coud be STM_PIN_DEFINE_SPEED_EXT instead of _EXT2 ?

Reply:
It's addressed to me?
change names up to you reasons,,

Reply:
I cannot edit your Pull request. The idea is to all review together your proposal and ask you to take also our feedback into account, so that your code will be in master branch of mbed-os in the end and everyone can benefit for this improvement. It's just a minor change, so if you do it all your target code will also be compatible with it when it gets merged

Reply:
STM_PIN_DEFINE_EXT already exist and used in some places of existing code

Reply:
Edits from maintainers are enabled for this pull request, Or, if I should make changes by own - what should I do?

Reply:
Please do changes as requested!